virttest/qemu_vm.py

```diff
@@ -4464,7 +4494,33 @@ def eject_cdrom(self, device, force=False):
         :param device: device ID;
         :param force: force eject or not;
         """
-        return self.monitor.eject_cdrom(device, force)
+        status_before = status_after = self.check_device_tray_opened(device)
+
+        qmp_monitors = [m for m in self.monitors if m.protocol == "qmp"]
+        for m in qmp_monitors:
+            m.clear_events()
+
+        out = self.monitor.eject_cdrom(device, force)
+
+        for m in qmp_monitors:
+            events = m.get_events()
```

Comment:
As QMP is asynchronous, it's better to call wait_for() here to wait event in the timeout.